docs/static/config-details.asciidoc

```diff
@@ -72,20 +82,70 @@ the higher percentage you can use.
 
 
 [[off-heap-size]]
-==== Setting the off-heap size
+===== Setting the off-heap size
 
 The operating system, persistent queue mmap pages, direct memory, and other processes require memory in addition to memory allocated to heap size.
-Keep the overall memory requirements in mind when you allocate memory.
+
+Internal JVM data structures, thread stacks, memory mapped files and direct memory for input/output (IO) operations are all parts of the off-heap JVM memory.
+Thread stacks zone contains the list of stack frames for each Java thread spinned by the JVM; each frame keeps the local arguments passed during method calls.
+Read on <<stacks-size>> if the size needs to be adapted to the processing needs.
+Memory mapped files are not part of the Logstash's process off-heap memory, but consume RAM when paging files from disk.
+These mapped files speed up the access to Persistent Queues pages, a performance improvement - or trade off - to reduce expensive disk operations such as read, write, and seek.
+Some network I/O operations also resort to in-process direct memory usage to avoid, for example, copying of buffers between network sockets.
+
+Keep these memory requirements in mind as you calculate your ideal memory allocation.
+
+[[memory-size-calculation]]
+===== Memory sizing
 
 By default, a JVM's off-heap direct memory limit is the same as the heap size. Check out <<plugins-inputs-beats-memory,beats input memory usage>>.
 Consider setting `-XX:MaxDirectMemorySize` to half of the heap size.
+Total JVM memory allocation must be estimated and is controlled indirectly using Java heap and direct memory settings.
+
+As you make your capacity calculations, keep in mind that the JVM can't consume the total amount of the host's memory available,
+as the Operating System and other processes will require memory too.
+
+For a {ls} instance with persistent queue (PQ) enabled on multiple pipelines, we could
+estimate memory consumption using:
+
+[source,text]
+-----
+pipelines number * (pipeline threads * stack size + 2 * PQ page size) + direct memory + Java heap
+-----
+
+NOTE: Each Persistent Queue requires that at least head and tail pages are present and accessible in memory.
+The default page size is 64 MB so each PQ requires at least 128 MB of heap memory, which can be a significant source
+of memory consumption per pipeline.
+
+Another component of memory consumption is the thread stack size, allocated per thread.
+Plugins, depending on their type (inputs, filters, and outputs), have different thread models.
+Every input plugin runs in its own thread and can potentially spawn others. For example, each JDBC input
+plugin launches a scheduler thread. Netty based plugins like TCP, Beats or HTTP input spawn a thread pool with 2 * number_of_cores threads.
+Output plugins may also start helper threads, such as a connection management thread for each
+{es} output instance.
+Every pipeline, then, has its own thread responsible to manage the pipeline lifecycle.
```

Comment:
I'd suggest moving this explanation to the off heap section, then the sizing section can focus just on which variables to use and how to mix them together

Reply:
Fixed by 28e4c19b36250618f1ba0e32091d0c9a33fe3c20